### DISCUSSION TITLE
Build options object based on configuration file at run/build time

### DISCUSSION THREAD

**Discussion by dcantatore** [nextauthjs/next-auth#983] [Help]
**Your question**
How can I build an options.providers array in real time?

**What are you trying to do**
I have a multi tenant setup and I would like to be able to build the options based on an object that has an array I would iterate over and push into the providers.options array. This works when the client side requests providers but the api part of the [...nextAuth] file says: 

```
API resolved without sending a response for /api/auth/session, this may result in stalled requests.
API resolved without sending a response for /api/auth/providers, this may result in stalled requests.
```
And when you try to use the providers they do no behave correctly.

**Reproduction**
I have my nextauth api file configure as shown in the example with the following changes
```

export default async (req, res) => {
    const tenantConfig = 'gets the tenant config file here';
    if (options.providers.length < tenantConfig.auth.length) {
        for (const authObj of tenantConfig.auth) {
            options.providers.push(Providers[authObj.type](authObj.values))
        }
    }
    NextAuth(req, res, options);
};
```

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [X] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [X] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by abisewski** [comment]
Hi!

Did you fix this "api/auth/session, this may result in stalled requests"?
